### ISSUE TITLE
Issues with Reading res1D file and mikeio.res1d QueryDataReach

### ISSUE THREAD

**Issue by awcherie** [DHI/mikeio#182]
**Is your feature request related to a problem? Please describe.**
Met issues when reading the res 1D files and using mikeio.res1d QueryDataReach
Reference: [https://github.com/DHI/mikeio/blob/master/notebooks/Res1D.ipynb](url)

**Issues**

1. **Data extracted via Res1D(file_path).read() is incomplete**
When reading the res1D file to dataframe, I realized the number of columns for each individual branch/reach in the dataframe does not tally with that as shown using the MIKE View. E.g. res1D file extracted using the python code only has 5 columns for 'Water Level', whereas res1D file extracted from MIKE View has 11 columns. This is the same issue for the 'Discharge' quantity. 

2. **Data header in dataframe is different from that of the chainage number**
res1D file to dataframe has the headers in the format 'Water Level: Branch: Column_number_count', which does not correspond to the format in MIKE View: 'Branch Chainage'. Hence, this makes it very tedious to manually match the result values to the correct headers, in order to extract the correct time series for individual chainages. 

3. **Data header extracted via Res1D(file_path).read() does not follow the order of ascending chainage**
res 1D file in MIKE View is in the order of, e.g. (Branch 736, Branch 744, Branch 1164, Branch 1184. etc). However, the read res1D to dataframe returns in the order of e.g. (Branch 1164 [1:0], Branch 1184 [1:1] , Branch 736 [1:35], Branch 744 [1:36], etc). 

**Summary of Issue 1 to 3**
![Summary_Screenshot](https://user-images.githubusercontent.com/48267739/107871784-ab8f7500-6edf-11eb-964d-f830b8772a42.PNG)

4. **mikeio.res1d QueryDataReach**
No data was returned for a valid branch/reach and chainage. Please advise. 
![Error_QueryDataReach](https://user-images.githubusercontent.com/48267739/107871927-f5c52600-6ee0-11eb-9ab2-220842d92bc4.PNG)

**Describe the solution you'd like**

1. To extract all information from the corresponding branch chainages, as per the chainages shown if were to process in MIKE View. 
2. To have the data column headers in the dataframe in the format of 'Branch Chainage' instead of Branch: Column_number_count) w.r.t. MIKE View. This will aid efficient extraction of results based on chainage. 
3. More detailed guide to using QueryDataReach 

**Describe alternatives you've considered**
-  Map the data header by using the values
- Using mikeio.res1d QueryDataReach to extract for specific reach and chainage

**Comment by rhaDHI** [comment]
@awcherie Thanks for reporting this. I assume the data your are looking at is the publicly available res1d file in this repo.

Will look at it next week unless someone else jumps in :-) Meanwhile, you could try to debug it and contribute with a solution your self. Would be happy to assist and review your first steps in your own branch. The res1d code is not that many lines https://github.com/DHI/mikeio/blob/master/mikeio/res1d.py . By inspecting this file I think you will realize how QueryDataReach works and maybe contribute with a few examples or doc string comments (in code or just here).

**Comment by rhaDHI** [comment]
Hi again @awcherie , I found some time to look into this. 

Please provide (a link to) the file you are using. Can be an anonymized subset as the actual values or names don't matter.

Issue 1: The code extracts all datasets and all data items in each set except GlobalData. So is the missing data part of the currently unsupported GlobalData set within the res1d file? Otherwise I need the file to debug through why the data is not there.

Issue 2: This is the default. You can change this by setting the argument 'put_chainage_in_col_name' in the Res1D constructor to True:
'''
class Res1D:
    def __init__(self, file_path=None, put_chainage_in_col_name=False):
'''

Issue 3: The dataframe columns are built in the order it appears in the res1d file. But you can reorganize the columns using pandas. Should not be more than a few lines. I will add this as default. 

Issue 4: Either there is no branch named "BRANCH1" or no chainage with number 20 in the data. I will add an exception stating this. This might be related to issue 1.

I hope this helps.

**Comment by awcherie** [comment]
Hi @rhaDHI, 

Thank you for taking the time to run through the errors. 

Pertaining to Issue #1 & #4, I had used my own result file to test. I could send the file to you for troubleshooting via email.

**Comment by rhaDHI** [comment]
@awcherie Thanks for the file via email. Here is an update:

Issue 1: Some time series were named incorrectly and missing chainage numbers. This happened for reaches that only have one element. Should be fixed now.

Issue 2: I changed the default to True for the argument 'put_chainage_in_col_name' in the Res1D constructor. Should solve this

Issue 3: I added alphabetical sorting, but this does not sort the chainage numbers in the string correctly in all cases. I will add a separate feature request for this.

Issue 4: The quantity in your res11 had white spaces between the strings, e.g. "Water Level" in stead of the mike1d standard "WaterLevel". I now compare the query quantity to the actual quantity names in the file instead of the standard list. So should be fixed now.

I created a PR currently in review that solves your issues. Should be in the next release :)
